src/_ert/forward_model_runner/cli.py

```diff
@@ -135,21 +143,71 @@ def main(args):
         ee_cert_path,
         experiment_id,
     )
+    reporter_queue: asyncio.Queue[Message] = asyncio.Queue()
+
+    done_flag = asyncio.Event()
+
+    forward_model_runner_task = asyncio.create_task(
+        ForwardModelRunner(jobs_data, reporter_queue=reporter_queue).run(
+            parsed_args.job
+        )
+    )
+    reporting_task = asyncio.create_task(
+        handle_reporting(reporters, reporter_queue, done_flag)
+    )
+
+    def handle_sigterm(*args, **kwargs):
+        nonlocal reporters, forward_model_runner_task
+        forward_model_runner_task.cancel()
+        for reporter in reporters:
+            reporter.cancel()
 
-    job_runner = ForwardModelRunner(jobs_data)
+    asyncio.get_running_loop().add_signal_handler(signal.SIGTERM, handle_sigterm)
 
-    for job_status in job_runner.run(parsed_args.job):
+    await forward_model_runner_task
+
+    done_flag.set()
+    await reporting_task
+
+
+async def handle_reporting(
+    reporters: Sequence[reporting.Reporter],
+    message_queue: asyncio.Queue[Message],
+    done: asyncio.Event,
+):
+    while not done.is_set() or not message_queue.empty():
```

Comment:
`not message_queue.empty()` looks dangerous. I think that this get handled in here:
`job_status = await asyncio.wait_for(message_queue.get(), timeout=2)` or ?

Reply:
I wanted it to process all the events in the queue before exiting, but I can rewrite it to be clearer ✏️